plugin/src/main/java/org/opensearch/ml/rest/RestMLExecuteAgentAction.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.rest;
+
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.settings.MLFeatureEnabledSetting;
+import org.opensearch.ml.common.transport.agent.MLExecuteAgentAction;
+import org.opensearch.ml.common.transport.agent.MLExecuteAgentRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+public class RestMLExecuteAgentAction extends BaseRestHandler {
+
+    private MLFeatureEnabledSetting mlFeatureEnabledSetting;
+
+    public RestMLExecuteAgentAction(MLFeatureEnabledSetting mlFeatureEnabledSetting) {
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(POST, "/_plugins/_ml/agents/{agent_id}/_execute"));
```

Comment:
Since we have the same route [here](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/rest/RestMLExecuteAction.java#L70), wouldn't this create a route conflict?